src/core/thread/mle_router.cpp

```diff
@@ -3135,49 +3135,34 @@ bool MleRouter::IsMinimalChild(uint16_t aRloc16)
 
 void MleRouter::RemoveNeighbor(Neighbor &aNeighbor)
 {
-    switch (mRole)
+    if (mRole == OT_DEVICE_ROLE_CHILD && &aNeighbor == &mParent)
     {
-    case OT_DEVICE_ROLE_DISABLED:
-    case OT_DEVICE_ROLE_DETACHED:
-        break;
+        BecomeDetached();
+    }
 
-    case OT_DEVICE_ROLE_CHILD:
-        if (&aNeighbor == &mParent)
+    if (!IsActiveRouter(aNeighbor.GetRloc16()))
+    {
+        if (aNeighbor.IsStateValidOrRestoring())
         {
-            BecomeDetached();
+            Signal(OT_NEIGHBOR_TABLE_EVENT_CHILD_REMOVED, aNeighbor);
         }
 
-        break;
+        aNeighbor.SetState(Neighbor::kStateInvalid);
```

Comment:
duplicate with [the last statement](https://github.com/openthread/openthread/pull/4234/files#diff-5a4f49e3956baf8690d055ae8a75a663R3169) 
could be removed?

Reply:
It was duplicate in the original code. 
I didn't dared not to change it.

Reply:
It was duplicate in the original code. 
I didn't dare to change it.

Reply:
I could not see the possible risk so far :)